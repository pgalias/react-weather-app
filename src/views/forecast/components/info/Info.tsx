import React, { FunctionComponent } from 'react';
import styles from './info.module.css';

type Props = {
  label: string;
  value: number;
  unit: string;
  icon?: JSX.Element;
};

export const Info: FunctionComponent<Props> = ({ label, value, unit, icon }) => (
  <p className={styles.info} role="listitem">
    <span className={styles.label}>{label}</span>
    <span className={styles.value}>{value}</span>
    <span className={styles.unit}>
      {unit}
      {icon}
    </span>
  </p>
);
