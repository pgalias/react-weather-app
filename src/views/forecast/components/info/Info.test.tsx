import React from 'react';
import { render, screen } from '@testing-library/react';
import { Info } from './Info';
import { ReactComponent as Icon } from '../../../../assets/img/down-arrow.svg';

describe('forecast::components::Info', () => {
  test('should render passed label, value and unit', () => {
    render(<Info label="foo" value={20} unit="m/s" />);

    expect(screen.getByRole('listitem')).toBeInTheDocument();
    expect(screen.getByText(/foo/i)).toBeInTheDocument();
    expect(screen.getByText(/20/i)).toBeInTheDocument();
    expect(screen.getByText(/m\/s/i)).toBeInTheDocument();
  });

  test('should render icon when its passed', () => {
    render(<Info label="bar" value={80} unit="%" icon={<Icon />} />);

    expect(screen.getByRole('listitem')).toMatchSnapshot();
  });
});
