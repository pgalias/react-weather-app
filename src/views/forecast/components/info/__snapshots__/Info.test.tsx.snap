// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forecast::components::Info should render icon when its passed 1`] = `
<p
  class="info"
  role="listitem"
>
  <span
    class="label"
  >
    bar
  </span>
  <span
    class="value"
  >
    80
  </span>
  <span
    class="unit"
  >
    %
    <svg>
      down-arrow.svg
    </svg>
  </span>
</p>
`;
